### ISSUE TITLE
Mark some Postgres types as `PhantomPinned`

### ISSUE THREAD

**Issue by workingjubilee** [tcdi/pgx#910]
Postgres types that have pointers into themselves should prevent themselves from being moved.

**Comment by eeeebbbbrrrr** [comment]
How come bindgen doesn't do this for us and do you or @thomcc think that's a thing it should do for us and should we give them a PR and would that be easy?

**Comment by workingjubilee** [comment]
For something like
```rust
struct Data {
    field: SomeData,
}

struct ComplexType {
    ptr: *mut Data,
    data: Data,
}

struct AnotherType {
    header: Metadata,
    data: Data,
}
```

Bindgen doesn't automatically know `ComplexType`'s `ptr` is pointing at `data` and not a `*mut Data` held by `AnotherType`.

**Comment by thomcc** [comment]
Yeah, it needs to be done manually. Or rather, a list of a types needs to be done manually -- if we have that, it becomes easy to insert something like `_marker: PhantomPinned` into that struct when generating bindings.

**Comment by thomcc** [comment]
I think this is slightly unnecessary, unless we want to use Pin aggressively. That said, it probably doesn't hurt.
